File: Java_Solutions/Solution5.java
public class Solution5 {
    // User input
    public static int fibonacci(int n) {
        if (n <= 0) return 0;
        if (n == 1) return 1;
        
        int prev = 0, curr = 1;
        for (int i = 2; i <= n; i++) {
            int next = prev + curr;
            prev = curr;
            curr = next;
        }
        return curr;
    }
    // End of user input

    static class Checker {
        public static void check(int n, int expected) {
            int result = fibonacci(n);
            if (result == expected) {
                System.out.printf("Test Case Passed: Result = %d, Expected = %d%n", 
                                result, expected);
            } else {
                System.out.printf("Failed Test Case: Result = %d, Expected = %d%n", 
                                result, expected);
            }
        }
    }

    public static void main(String[] args) {
        if (args.length != 2) {
            System.err.printf("Usage: java Solution5 <n> <expected>%n");
            System.exit(1);
        }

        int n = Integer.parseInt(args[0]);
        int expected = Integer.parseInt(args[1]);

        Checker.check(n, expected);
    }
} 

File: Solutions/C_Solutions/solution5.c
#include <stdio.h>
#include <stdlib.h>

int fibonacci(int n) {
    if (n <= 0) return 0;
    if (n == 1) return 1;
    return fibonacci(n - 1) + fibonacci(n - 2);
}

void Checker(int (*func)(int), int input, int expected) {
    int result = func(input);
    if (result == expected) {
        printf("Test Case Passed: Result = %d, Expected = %d\n", result, expected);
    } else {
        printf("Failed Test Case: Result = %d, Expected = %d\n", result, expected);
    }
}

int main(int argc, char *argv[]) {
    if (argc != 3) {
        fprintf(stderr, "Usage: %s <n> <expected>\n", argv[0]);
        return 1;
    }
    int n = atoi(argv[1]);
    int expected = atoi(argv[2]);
    Checker(fibonacci, n, expected);
    return 0;
} 